vespalib/src/vespa/vespalib/datastore/compaction_spec.h

```diff
@@ -0,0 +1,28 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#pragma once
+
+namespace vespalib::datastore {
+
+/*
+ * Class describing how to compact a compactable data structure.
+ *
+ * memory        - to reduce amount of "dead" memory
+ * address_space - to avoid running out of free buffers in data store
+ *                 (i.e. move data from small buffers to larger buffers)
+ */
+class CompactionSpec
+{
+    bool _compact_memory;
+    bool _compact_address_space;
+public:
+    CompactionSpec(bool compact_memory, bool compact_address_space)
+        : _compact_memory(compact_memory),
+          _compact_address_space(compact_address_space)
+    {
+    }
+    bool get_compact_memory() const noexcept { return _compact_memory; }
```

Comment:
As this function returns a bool, I think it reads better when removing the _get__ prefix. Applies to the function below as well.

Reply:
Fixed.